src/printer.js

```diff
@@ -3260,6 +3273,84 @@ function hasLeadingOwnLineComment(text, node) {
   return res;
 }
 
+function hasNakedLeftSide(node) {
+  return (
+    node.type === "BinaryExpression" ||
```

Comment:
This is the style used pretty consistently throughout the prettier codebase; I moved to a fake-set once (dictionary with values of `true`) and was asked to move back to this style. 

It might be possible that a Set-based style could be adopted in the future, but that'd probably best be done as a separate refactoring PR (with a preceding issue verifying that it's desired).